src/clj_sqs_extended/internal/receive.clj

```diff
@@ -32,44 +37,60 @@
                   (secs-between (-> loop-state :stats :started-at)
                                 now)))))
 
+(defn- stop-receive-loop
+  [loop-state]
+  (when (:running @loop-state)
+    (log/warnf "Terminating receive-loop for queue '%s' ..."
+               (:queue-name @loop-state))
+    (swap! loop-state assoc :running false)
+    (swap! loop-state assoc-in [:stats :stopped-at] (t/now))
+    (close! (:in-chan @loop-state))
+    (close! (:out-chan @loop-state)))
+  (:stats @loop-state))
+
 (defn- restart-receive-loop
   [sqs-ext-client loop-state receive-opts]
-  (log/infof "Restarting receive-loop for queue '%s'" (:queue-name @loop-state))
+  (log/infof "Restarting receive-loop for queue '%s' ..."
+             (:queue-name @loop-state))
   (let [old-in-chan (:in-chan @loop-state)]
     (swap! loop-state
            (fn [state]
              (-> state
-                 (assoc :in-chan (sqs/receive-message-channeled sqs-ext-client
-                                                                (:queue-name @loop-state)
-                                                                receive-opts))
+                 (assoc :in-chan
+                        (sqs/receive-message-channeled sqs-ext-client
+                                                       (:queue-name @loop-state)
+                                                       receive-opts))
                  (update-in [:stats :restart-count] inc)
                  (assoc-in [:stats :restarted-at] (t/now)))))
     (close! old-in-chan)))
 
-(defn- stop-receive-loop
-  [loop-state]
-  (when (:running @loop-state)
-    (log/infof "Terminating receive-loop for queue '%s'" (:queue-name @loop-state))
-    (swap! loop-state assoc :running false)
-    (swap! loop-state assoc-in [:stats :stopped-at] (t/now))
-    (close! (:in-chan @loop-state))
-    (close! (:out-chan @loop-state))
-    (:stats @loop-state)))
+(defn- error-might-be-recovered-by-restarting
+  [error]
+  (cond
+    (instance? ReflectiveOperationException error) false
+    (instance? RuntimeException error) false
+    (instance? UnknownHostException error) true
+    (instance? SocketException error) true
+    (instance? UnknownHostException error) true
+    (instance? HttpTimeoutException error) true
```

Comment:
can this be rewritten as one line each for sets of true and false errors? e.g. `(contains? #{UnknownHostException SocketException ...} (type error))`

Reply:
Totally agree, I'll go with your suggestion here.